host/sgx/enter.c

```diff
@@ -191,9 +190,16 @@ void oe_enter_sim(
 
     while (1)
     {
-        // Set GS and FS registers to values set by the ENCLU instruction upon
+        // Set FS registers to values set by the ENCLU instruction upon
         // entry to the enclave.
-        oe_set_gs_register_base((void*)(enclave->addr + sgx_tcs->gsbase));
+        // In Linux, the new value of FS persists until it is explicitly
+        // restored below. Windows however restores FS to the original value
+        // unexpectedly (say when the thread is suspended/resumed).
+        // This leads to access violations since features like stack-protector
+        // and thread-local storage use the FS register; but its value has been
+        // restored by Windows. To let the enclave chug along in simulation
+        // mode, we add signal handler that resets the FS register to the
+        // desired value.
```

Comment:
signal handler implies POSIX signals. Can you change the comment to explicitly note that its a vectored exception handler? Also since it isn't done in this file, add `in host/sgx/windows/exception.c` to the end.

Reply:
Nice suggestions. Done.